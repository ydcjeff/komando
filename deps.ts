export * from 'https://deno.land/std@0.127.0/flags/mod.ts';
