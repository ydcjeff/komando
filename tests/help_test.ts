import { assert, assertEquals, restoreAll, spyOn } from '../deps.ts';
import { komando } from '../mod.ts';
const { test } = Deno;

test('show default help', () => {
  const spy = spyOn(console, 'log');
  komando({
    name: 'help_test',
  }, ['-h']);
  assert(spy.called);
  assertEquals(spy.callCount, 2);
  assertEquals(
    spy.calls.flat().join('\n'),
    `
  Usage
    $ help_test [flags]

  Flags
    -h, --help    Show this message`,
  );
  restoreAll();
});

test('show command help', () => {
  const spy = spyOn(console, 'log');
  komando({
    name: 'help_test',
    commands: [
      {
        name: 'cmd',
        usage: '$ help_test cmd [flags]',
        commands: [],
        flags: {},
        args: {},
      },
    ],
  }, ['cmd', '--help']);
  assert(spy.called);
  assertEquals(spy.callCount, 2);
  assertEquals(
    spy.calls.flat().join('\n'),
    `
  Usage
    $ help_test cmd [flags]

  Flags
    -h, --help    Show this message`,
  );
  restoreAll();
});

test('show help + epilog', () => {
  const spy = spyOn(console, 'log');
  komando({
    name: 'help_test',
    epilog: '\n  Env Variables\n    CI: true',
  }, ['-h']);
  assert(spy.called);
  assertEquals(spy.callCount, 3);
  assertEquals(
    spy.calls.flat().join('\n'),
    `
  Usage
    $ help_test [flags]

  Flags
    -h, --help    Show this message

  Env Variables
    CI: true`,
  );
  restoreAll();
});
