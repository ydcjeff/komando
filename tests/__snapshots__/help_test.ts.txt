export const root_cmd_help = `
  Description
    Root command desc

  Usage
    Root command usage

  Commands
    sub1, s1                sub1 cmd desc

  Flags
    -p, --parent <dir>
        --flag-a <flagA>
    -h, --help              Show this message
    -V, --version           Show version info

  Args
    [argA]                  argA desc
    [argB...]               argB desc
    <argC...>               argC desc
    <argD>                  argD desc
    <argE,argE>             argE desc`;

export const sub_one_cmd_help = `
  Description
    sub1 cmd desc

  Aliases
    s1

  Usage
    $ root [command] [flags]

  SUBCOMMANDS
    subsub1
    subsub2
    subsub3

  SUBFLAGS
        --sub-flag-a <subFlagA>
        --sub-flag-b <subFlagB>
        --sub-flagc <subFlagc>

  Inherited Flags
    -h, --help                     Show this message
    -V, --version                  Show version info`;

export const minimal_cmd_help_only_flags = `
  Usage
    $ mini [flags]

  Flags
    -h, --help    Show this message`;

export const minimal_cmd_help_flags___args = `
  Usage
    $ mini [args] [flags]

  Flags
    -h, --help    Show this message

  Args
    <argA>`;

